### ISSUE TITLE
[Feature Request] Data Table - Accessible column headers

### ISSUE THREAD

**Issue by dmcknight26** [vuetifyjs/vuetify#13119]
### Problem to solve
Currently, when enabling sorting on columns in a data table they render with accessible text indicating if the column is currently sorted or not, as well as text indicating the column can be activated.  However the column headers themselves are nothing more than a `th` and a `span`, neither of which are accessible or selectable via keyboard.  Essentially a keyboard only user has no mechanism in which to toggle sorting despite what the accessible text implies.

### Proposed solution
I would propose either of the following:
* Convert sortable column headers into a properly selectable element, e.g. a button or anchor. Or at a minimum make something in the header cell tab-able so a keyboard user can interact with it.  My guess is that a button styled like a column header makes the most sense, however I'm not sure what the best option is here.
* Remove the click handler, hand cursor, and the accessible text currently on sortable columns, i.e. "Activate to sort [ascending|descending]".  Sorting would be handled externally and still displayed with an indicator next to the column name.  Perhaps this approach could be toggled via a new boolean property on the table?

In a nutshell I think the table either needs to lean into implementing the column headers in a more accessible way OR must provide a way to still display the current sort for external management while disabling any interaction on the header cells.

I did try using a completely custom header slot as an alternative, but using that approach disables the mobile functionality (where the table pivots), which I would really like to keep.  This is a known issue: https://github.com/vuetifyjs/vuetify/issues/10818.

**Comment by dmcknight26** [comment]
Related ticket: https://github.com/vuetifyjs/vuetify/issues/9164

**Comment by J-Sek** [comment]
Some progress with [27aaaf4](https://github.com/vuetifyjs/vuetify/commit/27aaaf41246aaaf3bed977c74d0b962daf0a5662) - sortable header cells are going to be accessible via keyboard

**Comment by msftedad** [comment]
Hi Team, Any update on this issue?

**Comment by J-Sek** [comment]
keyboard support shipped in v3.8.12 ([`27aaaf4`](https://github.com/vuetifyjs/vuetify/commit/27aaaf4)), `aria-sort` is going to appear in v4.1.0
